copy/copy.go

```diff
@@ -301,10 +295,43 @@ type diffIDResult struct {
 	err    error
 }
 
+// layerCopier allows copyLayer to cache mappings between srcInfo.Digest values and returned diffIDs, so that it can return diffIDs even when it's taking the ReapplyBlob shortcut.
+type layerCopier struct {
+	cachedDiffIDs map[digest.Digest]digest.Digest
```

Comment:
Okay, I think I've got that sorted.  The object is now called an `imageCopier`, since it also copies configuration blobs.